src/wallet/service_test.go

```diff
@@ -0,0 +1,430 @@
+package wallet
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/coin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	rand.Seed(time.Now().Unix())
+}
+
+func prepareWltDir() (string, func()) {
+	tmpDir := os.TempDir()
+	rand.Seed(time.Now().Unix())
+	n := rand.Int31n(100)
+	dir := fmt.Sprintf("%s%d/wallets", tmpDir, n)
+	return dir, func() {
+		os.RemoveAll(dir)
```

Comment:
If its putting the test data in /tmp we should leave it there and not call RemoveAll() in case we have another stupid bug that deletes $HOME

Reply:
I'm afraid of returning  same tmp wallet file, this will fail the test code

Reply:
oh sorry i misread, i thought you were using this:
https://golang.org/pkg/io/ioutil/#TempDir
https://golang.org/pkg/io/ioutil/#TempFile

if you use this there will be no problem